### ISSUE TITLE
ty incorrectly infers fields of SQLModel models as built-in types instead of sqlalchemy.orm attributes

### ISSUE THREAD

**Issue by ZivMBS** [astral-sh/ty#3421]
### Summary

Hey, I'm using SQLModel as an ORM for my db. I noticed 2 problems:

- Some operations on SQLModel models are flagged as `unresolved-attributes` by ty.
- ty incorrectly infers table fields as built-in types instead of SQLAlchemy's types.

Check this code:
```python
from sqlmodel import Field, SQLModel, select

class Item(SQLModel, table=True):
    id: int = Field(default=None, primary_key=True)
    name: str

stmt = select(Item).where(Item.id.in_([1, 2, 3]))  # <-- first problem, we get `Object of type `int` has no attribute `in_`` when using `ty check`
print(stmt)
print(type(Item.id))  # <-- Second problem - expected `<class 'sqlalchemy.orm.attributes.InstrumentedAttribute'>` but ty infers as int instead
if isinstance(Item.id, int):
    print("Ty expects this to run")
else:
    print("In reality this runs instead")
```

To run this you have to install the `SQLModel` package

### Version

ty 0.0.31

**Comment by nbouvrette** [comment]
Should be fixed in https://github.com/astral-sh/ruff/pull/25058
